extras/dhcpserver/include/dhcpserver.h

```diff
@@ -26,7 +26,19 @@ extern "C" {
    to a client.  Subsequent lease addresses are calculated by
    incrementing the final octet of the IPv4 address, up to max_leases.
 */
-void dhcpserver_start(const ip_addr_t *first_client_addr, uint8_t max_leases);
+void dhcpserver_start_on_if(const ip_addr_t *first_client_addr, uint8_t max_leases, struct netif *server_if);
+
+/* Start DHCP server
+
+   See @dhcpserver_start_on_if.
+
+   Selects the first netif in netif_list as the server_if
+ */
+static inline void dhcpserver_start(const ip_addr_t *first_client_addr, uint8_t max_leases)
+{
+	dhcpserver_start_on_if(first_client_addr, max_leases, NULL);
```

Comment:
I suggest just changing `dhcpserver_start` to take the `netif` and patching all the callers to pass in the appropriate netif which they can get using `sdk_system_get_netif`. Perhaps even place the `netif` as the first argument. That seems better than reaching into the lwip `netif_list` and taking the first element from that list and separated the dhcp server a little more from the internals of lwip.